core/src/Luna/IR/Expr.hs

```diff
@@ -369,6 +381,74 @@ impHub imps = mdo
     limps <- mapM (flip link t . unsafeRelayout) imps
     return t
 
+foreignImpSafety'
+    :: (ExprCons m ForeignImportSafety)
+    => ForeignImportType
+    -> m SomeExpr
+foreignImpSafety
+    :: (ExprCons m ForeignImportSafety)
+    => ForeignImportType
+    -> m (Expr ForeignImportSafety)
+foreignImpSafety' = generalize .: foreignImpSafety
+foreignImpSafety impType = mdo
+    newTerm <- expr $ Term.uncheckedForeignImportSafety impType
+    return newTerm
+
+foreignSymbolImp'
+    :: (ExprCons m ForeignSymbolImport)
+    => Expr a
+    -> Expr a
+    -> Name
+    -> Expr a
+    -> m SomeExpr
+foreignSymbolImp
+    :: (ExprCons m ForeignSymbolImport)
+    => Expr a
+    -> Expr a
+    -> Name
+    -> Expr a
+    -> m (Expr ForeignSymbolImport)
```

Comment:
I'd really prefer to keep it single line: 
```
foreignSymbolImp :: (ExprCons m ForeignSymbolImport)
    => Expr a -> Expr a -> Name -> Expr a -> m (Expr ForeignSymbolImport)
```

I understand why it was written this way. Keeping the 80 char limit is important, but let's also keep the vertical space minimal. Its equally important. We do not do any doc comments here and in such internal, self-describing functions we will not, so I'd stay with smaller code here.